test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -751,8 +751,148 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
+
+		AfterEach(func() {
+			cmd := []string{"rm", "-f", "/rootfs/var/roothome/create"}
+			nodes.ExecCommand(ctx, workerRTNode, cmd)
+			deletePod(ctx, guaranteedPod)
+			deletePod(ctx, secondPod)
+		})
+
+		It("[test_id: 74461] Verify that runc excludes the cpus used by guaranteed pod", func() {
+			guaranteedPod = getGuaranteedPod(ctx, workerRTNode)
+			secondPod = getNonGuaranteedPod(ctx, workerRTNode)
+			containerIDs := make([]string, 2)
+
+			var err error
+			containerIDs[0], err = pods.GetContainerIDByName(guaranteedPod, "test")
+			Expect(err).ToNot(HaveOccurred())
+			containerIDs[1], err = pods.GetContainerIDByName(secondPod, "test")
+			Expect(err).ToNot(HaveOccurred())
+			for _, containerID := range containerIDs {
+				path := fmt.Sprintf("/rootfs/var/run/containers/storage/overlay-containers/%s/userdata/config.json", containerID)
+				cmd := []string{"/bin/bash", "-c", fmt.Sprintf("cat %s >> /rootfs/var/roothome/create", path)}
+				nodes.ExecCommand(ctx, workerRTNode, cmd)
+			}
+			cmd := []string{"cat", "/rootfs/var/roothome/create"}
+			output, err := nodes.ExecCommand(context.TODO(), workerRTNode, cmd)
+			Expect(err).ToNot(HaveOccurred())
+
+			out := testutils.ToString(output)
+			hostnameRe := regexp.MustCompile(`"hostname":\s+"([^"]+)"`)
+			cpusRe := regexp.MustCompile(`"cpus":\s+"([^"]+)"`)
+
+			hostnameMatches := hostnameRe.FindAllStringSubmatch(out, -1)
+			cpusMatches := cpusRe.FindAllStringSubmatch(out, -1)
+			if len(hostnameMatches) == 0 || len(cpusMatches) == 0 {
+				Fail("Failed to extract hostname or cpus information")
+			}
+			uniqueCombinations := make(map[string]struct{})
+			zippedMatches := make([]map[string]string, 0)
+			for i := 0; i < len(hostnameMatches) && i < len(cpusMatches); i++ {
+				combination := fmt.Sprintf("%s-%s", hostnameMatches[i][1], cpusMatches[i][1])
+				if _, exists := uniqueCombinations[combination]; !exists {
+					uniqueCombinations[combination] = struct{}{}
+					zippedMatches = append(zippedMatches, map[string]string{
+						"hostname": hostnameMatches[i][1],
+						"cpus":     cpusMatches[i][1],
+					})
+				}
+			}
+
+			parseCPUs := func(cpuStr string) []int {
+				var cpus []int
+				for _, part := range strings.Split(cpuStr, ",") {
+					if strings.Contains(part, "-") {
+						bounds := strings.Split(part, "-")
+						min, _ := strconv.Atoi(bounds[0])
+						max, _ := strconv.Atoi(bounds[1])
+						for i := min; i <= max; i++ {
+							cpus = append(cpus, i)
+						}
+					} else {
+						val, _ := strconv.Atoi(part)
+						cpus = append(cpus, val)
+					}
+				}
+				return cpus
+			}
+
+			guaranteedPodCpus := parseCPUs(zippedMatches[0]["cpus"])
+			runcCpus := parseCPUs(zippedMatches[1]["cpus"])
+			overlapFound := false
+			for _, guaranteedCpu := range guaranteedPodCpus {
+				for _, runcCpu := range runcCpus {
+					if guaranteedCpu == runcCpu {
+						overlapFound = true
+						break
+					}
+				}
+				if overlapFound {
+					break
+				}
```

Comment:
I am confused by the above for loop,  the bug says we need to check if runc process is not using any of the cpus assigned to the pod, what exactly does this for loop achieve ?

Reply:
The loop here is to check if there is any common cpu present between the runc cpus and pod cpus after capturing them from config.json. we do not expect any common cpu number to be present.

Reply:
For that you don't need to use the loop, once you get the cpu's used by runtime(runc) ,   you  can use the existing method of getting the pod cpus and compare them using Expect. ?

Reply:
Can you review the new way, I tried with Intersection()